docs/examples-minimal.md

```diff
@@ -0,0 +1,220 @@
+programs](https://github.com/johnthagen/min-sized-rust) and are no longer
+specific to Wasmtime. For example the first thing that can be done is to
+optimize for size rather than speed via rustc's `s` optimization level.
+This uses Cargo's [environment-variable based configuration][cargo-env-config]
+via the `CARGO_PROFILE_RELEASE_OPT_LEVEL=s` environment variable to configure
+this.
+
+[cargo-env-config]: https://doc.rust-lang.org/cargo/reference/config.html#profile
+
+```
+$ export CARGO_PROFILE_RELEASE_OPT_LEVEL=s
+$ cargo build --release --no-default-features --features disable-logging
+$ ls -l ./target/release/wasmtime
+-rwxr-xr-x@ 1 root  root    6.8M Oct 18 08:40 target/release/wasmtime
+```
+
+Note that the size has increased here slightly instead of going down. Optimizing
+for speed-vs-size can affect a number of heuristics in LLVM so it's best to test
+out locally what's best for your embedding. Further examples below continue to
+pass this flag since by the end it will produce a smaller binary than the
+default optimization level of "3" for release mode. You may wish to also try an
+optimization level of "2" and see which produces a smaller build for you.
+
+After optimizations levels the next compilation setting to configure is
+Rust's "panic=abort" mode where panics translate to process aborts rather than
+unwinding. This removes landing pads from code as well as unwind tables from the
+executable.
+
+```
+$ export CARGO_PROFILE_RELEASE_OPT_LEVEL=s
+$ export CARGO_PROFILE_RELEASE_PANIC=abort
+$ cargo build --release --no-default-features --features disable-logging
+$ ls -l ./target/release/wasmtime
+-rwxr-xr-x@ 1 root  root    5.0M Oct 18 08:40 target/release/wasmtime
+```
+
+Next, if the compile time hit is acceptable, LTO can be enabled to provide
+deeper opportunities for compiler optimizations to remove dead code and
+deduplicate. Do note that this will take a significantly longer amount of time
+to compile than previously. Here LTO is configured with
+`CARGO_PROFILE_RELEASE_LTO=true`.
+
+```
+$ export CARGO_PROFILE_RELEASE_OPT_LEVEL=s
+$ export CARGO_PROFILE_RELEASE_PANIC=abort
+$ export CARGO_PROFILE_RELEASE_LTO=true
+$ cargo build --release --no-default-features --features disable-logging
+$ ls -l ./target/release/wasmtime
+-rwxr-xr-x@ 1 root  root    3.3M Oct 18 08:42 target/release/wasmtime
+```
+
+Similar to LTO above rustc can be further instructed to place all crates into
+their own single object file instead of multiple by default. This again
+increases compile times. Here that's done with
+`CARGO_PROFILE_RELEASE_CODEGEN_UNITS=1`.
+
+```
+$ export CARGO_PROFILE_RELEASE_OPT_LEVEL=s
+$ export CARGO_PROFILE_RELEASE_PANIC=abort
+$ export CARGO_PROFILE_RELEASE_LTO=true
+$ export CARGO_PROFILE_RELEASE_CODEGEN_UNITS=1
+$ cargo build --release --no-default-features --features disable-logging
+$ ls -l ./target/release/wasmtime
+-rwxr-xr-x@ 1 root  root    3.3M Oct 18 08:43 target/release/wasmtime
```

Comment:
Stripping debuginfo using `-Cstrip=debuginfo` should also help a fair bit, right?

Reply:
Oh excellent suggestion I forgot about that! Turns out the majority of the win from `-Zbuild-std` was stripping debuginfo